spec/translate.spec.js

```diff
@@ -0,0 +1,262 @@
+(function (factory) {
+  if (typeof define === 'function' && define['amd']) {
+    define(['chai', '../lib/translate', 'sinon', 'zurvan'], factory);
+  } else if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
+    factory(require('chai'), require('../lib/translate'), require('sinon'), require('zurvan'));
+  } else {
+    factory(chai, Translate, sinon, zurvan);
+  }
+} (function (chai, Translate, sinon, zurvan) {
+  'use strict';
+  const expect = chai.expect;
+  describe('Translate component', () => {
+    let instance, sandbox, resolveFetch, rejectFetch;
+
+    beforeEach(() => {
+      instance = new Translate();
+      sandbox = sinon.sandbox.create();
+      if (typeof window === 'undefined') {
+        global.window = global;
+      }
+      if (window.fetch === undefined) {
+        window.fetch = function () { };
+      }
+      if (window.alert === undefined) {
+        window.alert = () => { };
+      }
+      sandbox.stub(window, 'alert');
+      sandbox.stub(window, 'fetch', () => new Promise((resolve, reject) => {
+        resolveFetch = resolve;
+        rejectFetch = reject;
+      }));
+      return zurvan.interceptTimers().catch(() => { });
+    });
+    afterEach(() => {
+      sandbox.restore();
+      resolveFetch = null;
+      rejectFetch = null;
+      return zurvan.releaseTimers();
+    });
+
+    it('It should start with empty input ', () => {
+      expect(instance.input()).to.be.empty;
+    });
+
+    it('It should enable the Tranlste button', () => {
+      expect(instance.isBtnTranslateEnable()).to.equal(true);
+    });
+
+    it('It should start without selecting any language as source', () => {
+      expect(instance.src()).to.be.empty;
+    });
+
+    it('It should start without selecting any language as target', () => {
+      expect(instance.target()).to.be.empty;
+    });
+
+    it('It should not display an error to the user', () => {
+      expect(instance.errorMessage()).to.be.empty;
+    });
+
+    describe('When the user clicks the translate button before entering a word or choosing a language as source or as target', () => {
+      beforeEach(() => {
+        instance.translate();
+      });
+
+      it('Input should be empty', () => {
```

Comment:
I would say this test is not necessary at this stage. Same as the 2 tests below checking the languages are empty. However, a very relevant test would be to test that the translate button is still enabled so that the user can fix their error and try again.